monai/handlers/checkpoint_saver.py

```diff
@@ -0,0 +1,194 @@
+
+
+class CheckpointSaver:
+    """
+    CheckpointSaver acts as an Ignite handler to save checkpoint data into files.
+    It supports to save according to metrics result, epoch number, iteration number
+    and last model or exception.
+
+    Args:
+        save_dir (str): the target directory to save the checkpoints.
+        save_dict (dict): source objects that save to the checkpoint. examples::
+
+            {'network': net, 'optimizer': optimizer, 'engine', engine}
+
+        name (str): identifier of logging.logger to use, if None, defaulting to ``engine.logger``.
+        file_prefix (str): prefix for the filenames to which objects will be saved.
+        save_final (bool): whether to save checkpoint or session at final iteration or exception.
+        save_key_metric (bool): whether to save checkpoint or session according to key metric value.
+        key_metric_name (str): the name of key_metric in ignite metrics dictionary.
+            if None, use `engine.state.key_metric` instead.
+        key_metric_n_saved (int): save top N checkpoints or sessions sorted by the key metric.
+        epoch_save_interval (int): save checkpoint or session every N epochs.
+        epoch_n_saved (int): save latest N epochs' checkpoints or sessions, 'None' is to save all.
+        iteration_save_interval (int): save checkpoint or session every N iterations.
+        iteration_n_saved (int): save latest N iterations' checkpoints or sessions, 'None' is to save all.
+        convert_to_single_gpu (bool): if current model is trained by multi GPU and
+            want to save to single GPU model, need to save `object.module` instead.
+
+    Note:
+        CheckpointHandler can be used during training, validation or evaluation.
+        example of saved files:
+            checkpoint_iteration=400.pth
+            checkpoint_iteration=800.pth
+            checkpoint_epoch=1.pth
+            checkpoint_final_iteration=1000.pth
+            checkpoint_key_metric=0.9387.pth
+
+    """
+
+    def __init__(
+        self,
+        save_dir,
+        save_dict,
+        name=None,
+        file_prefix="",
+        save_final=False,
+        save_key_metric=False,
+        key_metric_name=None,
+        key_metric_n_saved=1,
+        epoch_save_interval=0,
+        epoch_n_saved=None,
+        iteration_save_interval=0,
+        iteration_n_saved=None,
+        convert_to_single_gpu=False,
+    ):
+        assert save_dir is not None, "must provide directory to save the checkpoints."
+        self.save_dir = save_dir
+        assert save_dict is not None and len(save_dict) > 0, "must provide source objects to save."
+        if convert_to_single_gpu:
+            for k, v in save_dict.items():
+                if hasattr(v, "module"):
+                    save_dict[k] = v.module
+        self.save_dict = save_dict
+        self.logger = None if name is None else logging.getLogger(name)
+        self.save_final = save_final
```

Comment:
On a style note I wouldn't store `self.save_final`, I would have `self.final_checkpoint` defined as `None` initially then set to an object is `save_final` is True. Further code would check if this attribute is `None` or not rather than checking `self.save_final`. I think one should always define attributes regardless of setup conditions, in this case `self.final_checkpoint` doesn't always exist which is confusing to someone inspecting the object's members.

Reply:
Sounds good, thanks for your suggestion.
I will update in next submit.

Reply:
updated in the latest patch.
Thanks.